lib/railties/all/railties.rbi

```diff
@@ -1,18 +1,1274 @@
 # typed: strong
 
 module Rails
-  sig {returns(Rails::Application)}
-  def self.application; end
-  
-  sig { returns(Pathname) }
-  def self.root; end
+  extend(::ActiveSupport::Autoload)
+
+  class << self
+    def app_class; end
+    def app_class=(_arg0); end
+
+    sig { returns(Rails::Application) }
+    def application; end
+
+    def application=(_arg0); end
+    def autoloaders; end
+
+    sig { returns(ActiveSupport::BacktraceCleaner) }
+    def backtrace_cleaner; end
+
+    sig { returns(ActiveSupport::Cache::Store) }
+    def cache; end
+
+    def cache=(_arg0); end
+    def configuration; end
+
+    sig { returns(ActiveSupport::StringInquirer) }
+    def env; end
+
+    def env=(environment); end
+    def gem_version; end
+    def groups(*groups); end
+    def initialize!(*_arg0, &_arg1); end
+    def initialized?(*_arg0, &_arg1); end
+
+    sig { returns(Logger) }
+    def logger; end
+
+    def logger=(_arg0); end
+    def public_path; end
+
+    sig { returns(Pathname) }
+    def root; end
+
+    sig { returns(String) }
+    def version; end
+  end
 end
 
-class Rails::Application
-  sig {params(blk: T.proc.bind(Rails::Application).void).void}
-  def configure(&blk); end
+class Rails::Application < ::Rails::Engine
+  def initialize(initial_variable_values = T.unsafe(nil), &block); end
+
+  def asset_precompiled?(logical_path); end
+  def assets; end
+  def assets=(_arg0); end
+  def assets_manifest; end
+  def assets_manifest=(_arg0); end
+  def build_middleware_stack; end
+
+  sig { returns(Rails::Application::Configuration) }
```

Comment:
I'm also opposed to this change for similar reasons as the others. The problem is that there are a _ton_ of configuration options in Rails, so while this is technically accurate, it means that any files that use `Rails.application.config.whatever` are going to now be marked `false` and fail to typecheck.

Unless we mapped every config method in rails (I wouldn't recommend it), I think this should remain untyped.

Reply:
We have a similar manual RBI in our repo where we type the configuration files with the custom methods we define.

Similar thing though, I’d be cool if you want update this to be unsigged with a comment explaining why.

Reply:
We have a custom config attribute `Rails.application.config.redis_manager` so we have the following RBI:
```
class Rails::Application::Configuration
  def redis_manager; end
  def redis_manager=(value); end
end
```

Reply:
I'd rather push it to the consumer, just because otherwise it causes type errors from Sorbet not understanding what `Rails.application.config` can be chained with